File: ts/src/test/base/test.sort.ts



import ccxt from '../../../ccxt.js';
import testSharedMethods from '../Exchange/base/test.sharedMethods.js';

function testSort () {
    // todo: other argument checks

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    const arr = [ 'b', 'a', 'c', 'd' ];
    const sortedArr = exchange.sort (arr);

    testSharedMethods.assertDeepEqual (exchange, undefined, 'sort', sortedArr, [
        'a',
        'b',
        'c',
        'd',
    ]);
}

export default testSort;

File: julia/Ccxt/test/base/test.sort.jl
using Test
using Ccxt
function testSort()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", sort!([], exchange), []);
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", sort!(["a"], exchange), ["a"]);
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", sort!(["a", "b", "c"], exchange), ["a", "b", "c"]);
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", sort!(["b", "a", "b", "c"], exchange), ["a", "b", "b", "c"]);
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", sort!(["b", "a", "c", "d"], exchange), ["a", "b", "c", "d"]);
    original = ["b", "a", "c"];
    sort!(original, exchange);
    assertDeepEqual(testSharedMethods, exchange, nothing, "sort", original, ["b", "a", "c"]);
end
